src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/ValidationServiceImpl.java

```diff
@@ -92,86 +66,64 @@
         this.lagLogicalPortRepository = persistenceManager.getRepositoryFactory().createLagLogicalPortRepository();
-        this.flowResourcesManager = new FlowResourcesManager(persistenceManager, flowResourcesConfig);
+        this.flowMeterRepository = persistenceManager.getRepositoryFactory().createFlowMeterRepository();
+        this.flowPathRepository = persistenceManager.getRepositoryFactory().createFlowPathRepository();
     }
 
     @Override
-    public ValidateRulesResult validateRules(SwitchId switchId, List<FlowEntry> presentRules,
-                                             List<FlowEntry> expectedDefaultRules) {
+    public ValidateRulesResult validateRules(SwitchId switchId, List<FlowSpeakerData> presentRules,
+                                             List<FlowSpeakerData> expectedRules) {
         log.debug("Validating rules on switch {}", switchId);
 
-        Set<Long> expectedCookies = getExpectedFlowRules(switchId);
-        return makeRulesResponse(expectedCookies, presentRules, expectedDefaultRules, switchId);
-    }
+        Set<Long> missingRules = new HashSet<>();
+        Set<Long> properRules = new HashSet<>();
+        Set<Long> excessRules = new HashSet<>();
+        Set<Long> misconfiguredRules = new HashSet<>();
 
-    private Set<Long> getExpectedServer42IngressCookies(SwitchId switchId, Collection<FlowPath> paths) {
-        return paths.stream()
-                .filter(path -> switchId.equals(path.getSrcSwitch().getSwitchId()))
-                .filter(path -> !path.isOneSwitchFlow())
-                .map(FlowPath::getCookie)
-                .map(FlowSegmentCookie::toBuilder)
-                .map(builder -> builder.type(CookieType.SERVER_42_FLOW_RTT_INGRESS))
-                .map(FlowSegmentCookieBuilder::build)
-                .map(CookieBase::getValue)
-                .collect(Collectors.toSet());
-    }
+        processRulesValidation(presentRules, expectedRules, missingRules, properRules, excessRules,
+                misconfiguredRules);
 
-    private Set<Long> getExpectedMirrorPointsCookies(SwitchId switchId, Collection<FlowPath> paths) {
-        Set<Long> cookies = new HashSet<>();
-        paths.forEach(path -> {
-            path.getFlowMirrorPointsSet().stream()
-                    .filter(mirrorPoints -> switchId.equals(mirrorPoints.getMirrorSwitchId()))
-                    .forEach(mirrorPoints -> {
-                        Collection<FlowMirrorPath> flowMirrorPaths = mirrorPoints.getMirrorPaths();
-                        if (!flowMirrorPaths.isEmpty()) {
-                            cookies.add(path.getCookie().toBuilder().mirror(true).build().getValue());
-                        }
-                    });
-        });
-        return cookies;
+        if (!missingRules.isEmpty() && log.isErrorEnabled()) {
+            log.error("On switch {} the following rules are missed: {}", switchId,
```

Comment:
Is it correct to log the operation result (even so unexpected) as errors?

Reply:
Lowered to warn.